fennel.lua

```diff
@@ -2868,6 +2804,33 @@ that argument name begins with ?."
     ;; many values as we ever match against in the clauses.
     (list (sym :let) [vals val]
           (match-condition vals clauses))))
+ :include
+ (fn include [module-name]
+   "Load given module and use its contents as macro definitions in currentscope.
+Macro module should return a table of macro functions with string keys."
+   (assert (= :string (type module-name))
+           "module name must resolve to a string literal")
+   (let [resolve            #(let [path (searchpath $ fennel.path)]
+                               (if path [path true]
+                                        [(searchpath $ package.path) false]))
+         [mod-path fennel?] (resolve module-name)
+         scope           (get-scope)
+         compile         (if fennel? fennel.compileString #$)
+         code            (if (and mod-path (not (. scope.includes module-name)))
+                             (: (io.open mod-path) :read :*a))
+         lua-code        (if code (compile code))
+         found           (or (. scope.includes module-name) lua-code)
+         includer        `(when (and (not (. package.preload ,module-name))
+                                     (= nil (. package.loaded ,module-name)))
+                                (tset package.preload ,module-name
+                                      ((or loadstring load) ,lua-code)))]
+     ; if requireAsInclude is enabled when unresolvable, fallback to require
+     (assert (or found (get-option scope :requireAsInclude))
+             (.. "could not find module " module-name))
+     (local out `(do ((values require) ,module-name)))
```

Comment:
Can you add a comment about why `(values require)` is necessary here? I think I understand it now, but I won't if I come back to it later. =)

Reply:
Will do (if we end up keeping this form instead of using bakpakin's diff below)! That's actually what the comment at the top of this snippet was supposed to be account for, but I can see why it would be unclear due to both the position and wording..